.github/workflows/update-deps.yml

```diff
@@ -23,9 +23,9 @@ jobs:
           ref: ${{ github.event.inputs.ref }}
       - name: set branches output
         id: branches
-        # outputs the two most recent `release-x.y` branches plus `main` as JSON
+        # outputs the second to most recent `release-x.y` branches plus `main` as JSON
         run: |
-          echo "branches=$(git branch -a | grep -E "remotes/origin/(main|release-)" | sed 's/  remotes\/origin\///' | sort -V | tail -2 | jq -R -s -c 'split("\n") | map(select(length > 0)) | . + ["main"]')" >> $GITHUB_OUTPUT
+          echo "branches=$(git branch -a | grep -E "remotes/origin/(main|release-)" | sed 's/  remotes\/origin\///' | sort -V | tail -2 | head -1 | jq -R -s -c 'split("\n") | map(select(length > 0)) | . + ["main"]')" >> $GITHUB_OUTPUT
```

Comment:
previously, it would open a PR for:
* release-0.8
* release-0.9
* main

because `main` will be merged into `0.9` there is no point in updating dependencies there. Hence only `n-1` and `main` will be selected here.